src/ILCompiler.Compiler/src/Compiler/CompilationModuleGroup.cs

```diff
@@ -81,6 +81,13 @@ private void AddWellKnownTypes(IRootingServiceProvider rootProvider)
             {
                 rootProvider.AddCompilationRoot(stringType, "String type is always generated");
             }
+
+            var coreLib = _typeSystemContext.GetModuleForSimpleName("System.Private.Corelib");
+
+            // Adding for CPP codeGen
+            var eeTypeType = coreLib.GetKnownType("Internal.Runtime", "EEType");
+            rootProvider.AddCompilationRoot(eeTypeType, "Enables C++ Code Generation");
```

Comment:
Can you help me find the place where CppCodegen is aware of the `Internal.Runtime.EEType` type? I can see it hardcoding `MethodTable*` and `RawEEType*`, but not the common managed reader implementation.